src/audio/google/google_rtc_audio_processing.c

```diff
@@ -392,18 +392,36 @@ static int google_rtc_audio_processing_init(struct processing_module *mod)
 #if CONFIG_IPC_MAJOR_4
 	const struct ipc4_base_module_extended_cfg *base_cfg = md->cfg.init_data;
 	struct ipc4_input_pin_format reference_fmt, output_fmt;
-	const size_t size = sizeof(struct ipc4_input_pin_format);
+	size_t in_fmt_size = sizeof(struct ipc4_input_pin_format);
+	size_t out_fmt_size = sizeof(struct ipc4_output_pin_format);
+	size_t size;
 
 	cd->config.base_cfg = base_cfg->base_cfg;
 
 	/* Copy the reference format from input pin 1 format */
-	memcpy_s(&reference_fmt, size,
-		 &base_cfg->base_cfg_ext.pin_formats[size], size);
-	memcpy_s(&output_fmt, size,
-		 &base_cfg->base_cfg_ext.pin_formats[size * GOOGLE_RTC_NUM_INPUT_PINS], size);
+	memcpy_s(&reference_fmt, in_fmt_size,
+		 &base_cfg->base_cfg_ext.pin_formats[in_fmt_size], in_fmt_size);
+	memcpy_s(&output_fmt, in_fmt_size,
+		 &base_cfg->base_cfg_ext.pin_formats[in_fmt_size * GOOGLE_RTC_NUM_INPUT_PINS],
+		 in_fmt_size);
 
 	cd->config.reference_fmt = reference_fmt.audio_fmt;
 	cd->config.output_fmt = output_fmt.audio_fmt;
+
+	/* save the base config extension */
+	size = base_cfg->base_cfg_ext.nb_input_pins * in_fmt_size +
+		base_cfg->base_cfg_ext.nb_output_pins * out_fmt_size;
```

Comment:
nitpick: you could have
```
	size = base_cfg->base_cfg_ext.nb_input_pins * in_fmt_size +
		base_cfg->base_cfg_ext.nb_output_pins * out_fmt_size +
		sizeof(struct ipc4_base_module_cfg_ext);
```

Reply:
@lyakh , no , see line 416, extra struct size are added when rzalloc

Reply:
@btian1 that's exactly why I'm proposing that